kong/runloop/plugins_iterator.lua

```diff
@@ -234,21 +251,19 @@ end
 -- @function load_configuration_through_combos
 -- @tparam table ctx A table containing the context information, including route, service, and authenticated_consumer.
 -- @tparam table combos A table containing configuration data indexed by compound keys.
--- @tparam table plugin A table containing plugin information, including the handler with no_route, no_service, and no_consumer rules.
 -- @treturn any|nil The configuration corresponding to the best matching combination, or 'nil' if no configuration is found.
 ---
-local function load_configuration_through_combos(ctx, combos, plugin)
-
-    -- Filter out route, service, and consumer based on the plugin handler rules and get their ids
-  local route_id = (ctx.route and not plugin.handler.no_route) and ctx.route.id or nil
-  local service_id = (ctx.service and not plugin.handler.no_service) and ctx.service.id or nil
-  -- TODO: kong.client.get_consumer() for more consistency. This might be an exception though as we're aiming for raw speed instead of compliance.
-  local consumer_id = (ctx.authenticated_consumer and not plugin.handler.no_consumer) and ctx.authenticated_consumer.id or nil
+local function load_configuration_through_combos(ctx, combos)
+  -- Filter out route, service, and consumer based on the plugin handler rules and get their ids
+  local route_id = ctx.route and ctx.route.id or nil
+  local service_id = ctx.service and ctx.service.id or nil
+  local consumer_id = ctx.authenticated_consumer and ctx.authenticated_consumer.id or nil
```

Comment:
This field lost the `no_consumer` validation. Is this intended?

Reply:
I don't recall what I was doing here. Perhaps it was related to getting rid of global here or something. These are performance optimizations, not validations?